light_curves/light_curve_classifier.md

```diff
@@ -319,45 +383,105 @@ def make_zero_lc(oid, band, label):
+                zerosingle = make_zero_light_curve(oid, band, label)
+                #keep track of these empty light curces in a list
+                zerosingle_list.append(zerosingle)
+    
+    #turn the empty light curves into a dataframe
+    df_empty = pd.DataFrame(zerosingle_list)
+    # df_empty has one row per dict. time,flux, and err columns store arrays.
+    # "explode" the dataframe to get one row per light curve point. time, flux, and err columns will now store floats.
+    df_empty = df_empty.explode(["time", "flux","err"], ignore_index=True)
+    df_empty = df_empty.astype({col: "float" for col in ["time", "flux", "err"]})
+
+
+    #now put the empty light curves back together with the main light curve dataframe
+    zeros_df_lc = pd.concat([df_lc, df_empty])
+
+    return(zeros_df_lc)
+```
+
+```{code-cell} ipython3
+def missingdata_drop_bands(df_lc, verbose = False):
+    """
+    Drop bands with the most missing data and objects without all remaining bands so that there is no missing data going forward.
+       
+    Parameters
+    ----------
+    df_lc: Pandas dataframe with light curve info
+    
+    verbose: bool
+    
+    Returns
+    --------
+    df_lc: MultiIndexDFObject with all  light curves
+        
+    """
+
+    #try instead to require that all objects have bands = df_lc.band.unique()
+
+    #first drop the bands not included from all objects
+    bands_to_drop = ['zi', 'Gaia g', 'Gaia bp', 'Gaia rp']
+    drop_df_lc = df_lc[~df_lc["band"].isin(bands_to_drop)]
     
-    #if it is not the complete list, ie some bandnames are missing:                            
-    if len(missing) > 0:
+    #now a list of all remaining bands
+    all_bands = drop_df_lc.band.unique()
     
-        #make new dataframe for this object with zero flux and err values
-        for band in missing:
-            label = str(singleoid.label.unique().squeeze())
-            zerosingle = make_zero_lc(oid, band, label)
-            #keep track of these empty light curces in a list
-            zerosingle_list.append(zerosingle)
+    if verbose:
+        #how many objects did we start with?
+        print(drop_df_lc.groupby(["objectid"]).ngroups, "n objects before removing missing band data")
+
+    # Identify objects with all bands
+    complete_objects = drop_df_lc.groupby('objectid')['band'].apply(lambda x: set(x) == set(all_bands))
 
-            
-#turn the empty light curves into a dataframe
-df_empty = pd.DataFrame(zerosingle_list)
-# df_empty has one row per dict. time,flux, and err columns store arrays.
-# "explode" the dataframe to get one row per light curve point. time, flux, and err columns will now store floats.
-df_empty = df_empty.explode(["time", "flux","err"], ignore_index=True)
-df_empty = df_empty.astype({col: "float" for col in ["time", "flux", "err"]})
+    # Filter the DataFrame based on complete objects
+    filter_df_lc = drop_df_lc[drop_df_lc['objectid'].isin(complete_objects[complete_objects].index)]
```

Comment:
I'm a bit confused about `complete_objects[complete_objects].index` here, but I think the result is the same as `complete_objects.index`? If so, and if you accept the above suggestion, then this should change to:

```suggestion
    filter_df_lc = drop_df_lc[drop_df_lc['objectid'].isin(complete_objects)]
```

But even looking at that suggestion, I'm confused about what the code is actually doing. I'm only reading, not actually running code, so I can't check for sure. This whole function is fine right now the way it is. So you can make changes if they make sense to you, or leave it as-is.

Reply:
I'm going to leave this whole function alone and let us come back to it at a later date.